xapian-applications/omega/handler_tesseract.cc

```diff
@@ -0,0 +1,85 @@
+/** @file handler_tesseract.cc
+ * @brief Extract text from Images using tesseract.
+ */
+/* Copyright (C) 2019 Bruno Baruffaldi
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+#include <config.h>
+#include "handler.h"
+
+#include <tesseract/baseapi.h>
+#include <leptonica/allheaders.h>
+
+using namespace std;
+using namespace tesseract;
+
+TessBaseAPI* ocr = NULL;
+
+static bool
+clear_text(string& dump, const char* text)
+{
+    bool k = false;
+    if (text) {
+	for (int i = 0; text[i] != '\0'; ++i) {
+	    if (!isspace(text[i]) || (i && !isspace(text[i - 1]))) {
+		dump.push_back(text[i]);
+		k |= isalnum(text[i]);
```

Comment:
This will only fire for ASCII alphanumerics, and so will work badly for text in other alphabets.

Reply:
Okay, I didn't realize of that.
I am going to change it for `!isspace`, but I am not sure is that is a good solution.
Here I am trying to detect if tesseract was able to extract some useful text and not just a bunch of spaces and punctuation characters without sense. Maybe we can use [isalnum](https://en.cppreference.com/w/cpp/locale/isalnum) or [iswalnum](https://en.cppreference.com/w/cpp/string/wide/iswalnum).

Reply:
The text will be UTF-8 not wide characters, so `iswalnum()` isn't directly helpful.

I don't think you need to worry about an "all whitespace" document here - omindex will notice that without your help and handle it appropriately.

If tesseract commonly returns some non-alphanumeric junk text for certain images then that would be good to handle, but I think it might be better not to try to overthink this up front, but instead get this working without such special handling and then test it on a variety of images which aren't scans and see what you get.  I tested on a handful here, and I seem to get either SPACE LINEFEED FORMFEED, or something which contains at least one ASCII letter.

Reply:
Okay, I will remove this check.